app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -105,7 +108,8 @@ class BrowserFragment : Fragment(), BackHandler {
                 when (it) {
                     is SearchAction.ToolbarTapped -> Navigation.findNavController(toolbar)
                         .navigate(BrowserFragmentDirections.actionBrowserFragmentToSearchFragment(
-                            requireComponents.core.sessionManager.selectedSession?.id
+                            requireComponents.core.sessionManager.selectedSession?.id,
+                            (activity as HomeActivity).browsingModeManager.isPrivate
```

Comment:
I suppose it's not a big deal and we don't have to fix it here, but in general it's considered not great to reference fields on the parent activity by casting. Fragments are designed to be reusable UI in theory to allow for new use cases later. The proper method is creating an interface between the two, which comes with an intense amount of boilerplate, or using a Architecture Components ViewModel.